camera.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2025 PHYTEC Messtechnik GmbH
+# SPDX-License-Identifier: Apache-2.0
+# Author: Martin Schwan <m.schwan@phytec.de>
+
+import cv2 as cv
+import glob
+import os
+import subprocess
+
+class Camera():
+    def __init__(self):
+        self.cameraDevice = None
+        self.colorConversionCode = None
+        self.apiPreference = None
+        self.videoCapture = cv.VideoCapture()
+
+    def open(self, filename):
+        if self.apiPreference is None:
+            raise AttributeError('API preference must be set before opening '
+                                 'video capture device!')
+        self.videoCapture.open(filename, self.apiPreference)
+        if not self.videoCapture.isOpened():
+            raise ValueError(f'Failed opening video capture device "{filename}"!')
+
+    def convert_frame_color(self, frame):
+        if self.colorConversionCode is None:
+            raise AttributeError('Color conversion code must be set before '
+                                 'converting frame colors!')
+        return cv.cvtColor(frame, self.colorConversionCode)
+
+class CameraUSB(Camera):
+    def __init__(self):
+        super().__init__()
+        self.colorConversionCode = cv.COLOR_BGR2RGB
+        self.apiPreference = cv.CAP_ANY
+
+    def open(self, filename=1):
+        super().open(filename)
+
+class CameraVM016(Camera):
+    def __init__(self):
+        super().__init__()
+        self.colorConversionCode = cv.COLOR_BAYER_GB2RGB
+        self.apiPreference = cv.CAP_GSTREAMER
+
+        if cv.getBuildInformation().find('GStreamer') < 0:
+            raise ValueError('This version of OpenCV does not support GStreamer!')
+
+    def open(self, filename=None):
+        if filename is None:
+            cameraDevices = glob.glob('/dev/cam-csi[0-9]*')
```

Comment:
There can only be `/dev/cam-csi1` and/or `/dev/cam-csi2` since there are only two CSI interfaces on the Pollux.
Also this demo does only support the CSI1 interface so far, see further bellow the `setup-pipeline-csi1` call. So I think it makes more sense to simply set the device to `/dev/cam-csi1` since it always points to the camera subdevice which is connected to the CSI1 interface.

Reply:
Yeah, this section is not ideal, I agree on that. This is probably a poor attempt from my side to automatically select an available interface.

I think I'll change it back to the static strings and remove the globbing.

My plan is to eventually make a more sophisticated detection of available devices and let the user set the desired interface through the GUI. We'll see how far that goes, but that's something for another PR.